utils/mgmt_system/kubernetes.py

```diff
@@ -0,0 +1,208 @@
+import os
+from conf import cfme_data, credentials
+from utils.rest_client import ContainerClient
+from collections import namedtuple
+
+"""
+Related yaml structures:
+
+[cfme_data]
+container_managers:
+    kubernetes:
+        name: My kubernetes
+        type: kubernetes
+        hostname: 10.12.13.14
+        port: 6443
+        credentials: kubernetes
+        authenticate: true
+        rest_protocol: https
+
+[credentials]
+kubernetes:
+    username: admin
+    password: secret
+    token: mytoken
+"""
+
+# Below are simplified representations of CFME k8s objects that should cover most of our needs
+Container = namedtuple('Container', ['name', 'cg_name', 'image'])
+Image = namedtuple('Image', ['name', 'tag', 'id'])
+# Images are equal if their tags match
+Image.__eq__ = lambda self, other: self.tag == other.tag
```

Comment:
I was thinking about this and we can get rid of this monkey patching:

```python
class Image(namedtuple("Image", ["name", "tag", "id"])):  # namedtuple creates a class, so it can be inherited!
    def __eq__(self, other):
        return self.tag == other.tag

In [10]: i = Image("asdf", "tag", 1)

In [11]: j = Image("asdf", "tag2", 1)

In [13]: i == j
Out[13]: False

In [14]: k = Image("asdf", "tag", 1)

In [15]: i == k
Out[15]: True
```

Not sure what would Mr. @seandst  say on this usage of namedtuples :)